store/dataStore.ts

```diff
@@ -1,17 +1,33 @@
 import { create } from 'zustand';
+import { produce } from 'immer';
+import { IUser } from '@/api/IapiFunctions';
 
-interface Example {
-    id: number;
-    name: string;
+interface State {
```

Comment:
Remove in favor of the changes in IapiFunctions.ts

Reply:
this will cause issues down the line.

State will be a collection of different state objects (not just IUser)

I recommend keeping this as is but move to the IDataStore.ts file

Reply:
❌ Did not fix per our conversation